File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/termo/UltimaRemuneracaoProxy.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.termo;

import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.negocio.dominio.termo.ParametroDoTermo;
import br.jus.trt8.pjecalc.negocio.dominio.termo.Termo;
import java.math.BigDecimal;

public class UltimaRemuneracaoProxy
implements Termo {
    private static final long serialVersionUID = 996548814284121647L;

    @Override
    public BigDecimal resolverValor(ParametroDoTermo parametro) {
        return parametro.getValorUltimaRemuneracaoDoCalculo();
    }

    public String toString() {
        return Utils.formatarNumero(this.resolverValor(null));
    }
}

File: src/lib/pjecalc/core/dominio/termo/ultima-remuneracao-proxy.ts
/**
 * Porte 1:1 de UltimaRemuneracaoProxy.java (24 linhas).
 * Delega para Calculo.getValorUltimaRemuneracao.
 *
 * Ref: pjecalc-fonte/.../dominio/termo/UltimaRemuneracaoProxy.java
 */
import type Decimal from 'decimal.js';
import type { Termo } from './termo';
import type { ParametroDoTermo } from './parametro-do-termo';

export class UltimaRemuneracaoProxy implements Termo {
  resolverValor(parametro: ParametroDoTermo): Decimal {
    return parametro.getValorUltimaRemuneracaoDoCalculo();
  }
}
